src/type/definition.js

```diff
@@ -66,6 +67,130 @@ export function assertType(type: mixed): GraphQLType {
+}
+
+declare function isObjectType(type: mixed): boolean %checks(type instanceof
+  GraphQLObjectType);
+// eslint-disable-next-line no-redeclare
+export function isObjectType(type) {
+  return instanceOf(type, GraphQLObjectType);
+}
+
+export function assertObjectType(type: mixed): GraphQLObjectType {
+  invariant(
+    isObjectType(type),
+    `Expected ${String(type)} to be a GraphQL Object type.`,
+  );
+  return type;
+}
+
+declare function isInterfaceType(type: mixed): boolean %checks(type instanceof
+  GraphQLInterfaceType);
+// eslint-disable-next-line no-redeclare
+export function isInterfaceType(type) {
+  return instanceOf(type, GraphQLInterfaceType);
+}
+
+export function assertInterfaceType(type: mixed): GraphQLInterfaceType {
+  invariant(
+    isInterfaceType(type),
+    `Expected ${String(type)} to be a GraphQL Interface type.`,
+  );
+  return type;
+}
+
+declare function isUnionType(type: mixed): boolean %checks(type instanceof
+  GraphQLUnionType);
+// eslint-disable-next-line no-redeclare
+export function isUnionType(type) {
+  return instanceOf(type, GraphQLUnionType);
+}
+
+export function assertUnionType(type: mixed): GraphQLUnionType {
+  invariant(
+    isUnionType(type),
+    `Expected ${String(type)} to be a GraphQL Union type.`,
+  );
+  return type;
+}
+
+declare function isEnumType(type: mixed): boolean %checks(type instanceof
+  GraphQLEnumType);
+// eslint-disable-next-line no-redeclare
+export function isEnumType(type) {
+  return instanceOf(type, GraphQLEnumType);
+}
+
+export function assertEnumType(type: mixed): GraphQLEnumType {
+  invariant(
+    isEnumType(type),
+    `Expected ${String(type)} to be a GraphQL Enum type.`,
+  );
+  return type;
+}
+
+declare function isInputObjectType(type: mixed): boolean %checks(type instanceof
+  GraphQLInputObjectType);
+// eslint-disable-next-line no-redeclare
+export function isInputObjectType(type) {
+  return instanceOf(type, GraphQLInputObjectType);
+}
+
+export function assertInputObjectType(type: mixed): GraphQLInputObjectType {
+  invariant(
+    isInputObjectType(type),
+    `Expected ${String(type)} to be a GraphQL Input Object type.`,
+  );
+  return type;
+}
+
+declare function isListType(type: mixed): boolean %checks(type instanceof
+  GraphQLList);
+// eslint-disable-next-line no-redeclare
+export function isListType(type) {
+  return instanceOf(type, GraphQLList);
+}
+
+export function assertListType(type: mixed): GraphQLList<any> {
+  invariant(
+    isListType(type),
+    `Expected ${String(type)} to be a GraphQL List type.`,
+  );
+  return type;
+}
+
+declare function isNonNullType(type: mixed): boolean %checks(type instanceof
+  GraphQLNonNull);
+// eslint-disable-next-line no-redeclare
+export function isNonNullType(type) {
+  return instanceOf(type, GraphQLNonNull);
+}
+
+export function assertNonNullType(type: mixed): GraphQLNonNull<any> {
+  invariant(
+    isNonNullType(type),
+    `Expected ${String(type)} to be a GraphQL Non-Null type.`,
```

Comment:
@leebyron Maybe it worth to say something like `to be wrapped in`?
Same goes for List.